### PR TITLE
Log crash during mounting

### PR DESCRIPTION
**Pull request by Tulsishah**
Description
Background, with logfile - Panic message printed in Logfile
Background, without logfile - Panic message printed in Syslog
Foreground, with logfile - Panic message printed in Logfile
Foreground, without logfile - Panic message printed on Console

Fatal print simple statement with os.Exit(1)
E.g. I0517 10:38:22.851246 Panic: Test

Testing
Manual - Tested manually by putting panic message before mounting
UnitTest - N/A
IntegrationTest - N/A

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -423,7 +423,18 @@ func run() (err error) {
 	return
 }
 
+// Handle panic if the crash occurs during mounting.
+func handlePanic() {
+	// Detect if panic happens in main go routine.
+	a := recover()
+	if a != nil {
+		logger.Info("Panic: ", a)
```
Comment (vadlakondaswetha) [inline main.go:431]: why is panic an info message?

Comment (Tulsishah) [reply main.go:431]: As panic is will do exit code 1, before that we are printing the cause of panic.

Comment (vadlakondaswetha) [reply main.go:431]: message type - info or error represents the type of log. Yes panic will crash the process, that doesn't mean we print the source of panic as info message. 
https://stackoverflow.com/questions/2031163/when-to-use-the-different-log-levels

FYI, here you made panic log as fatal: https://stackoverflow.com/questions/2031163/when-to-use-the-different-log-levels

Comment (Tulsishah) [reply main.go:431]: Yes I thought that but [here](https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/internal/fs/wrappers/error_mapping.go#L93) em errormapping logger have Fatal method but our [logger](https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/internal/logger/logger.go) does not have Fatal method.

Should I add that method in our logger in proceed ?

Comment (vadlakondaswetha) [reply main.go:431]: yes

File: internal/logger/logger.go [hunk 2]
```diff
@@ -133,6 +133,10 @@ func Info(v ...interface{}) {
 	defaultInfoLogger.Println(v...)
 }
 
+func Fatal(v ...interface{}) {
```
Comment (raj-prince) [inline internal/logger/logger.go:136]: Can you please write a one-liner as like in the other exported method?

Also, can you please update the description how fatal logs look like?

File: main.go [hunk 3]
```diff
@@ -423,7 +423,18 @@ func run() (err error) {
 	return
 }
 
+// Handle panic if the crash occurs during mounting.
+func handlePanic() {
```
Comment (raj-prince) [inline main.go:427]: We can rename this method as it looks like so generic - handlePanicWhileMounting().

**Review by raj-prince — APPROVED** [review]

File: internal/logger/logger.go [hunk 4]
```diff
@@ -133,6 +133,11 @@ func Info(v ...interface{}) {
 	defaultInfoLogger.Println(v...)
 }
 
+// Fatal calls the default info logger to print the message followed by a call to os.Exit(1).
```
Comment (vadlakondaswetha) [inline internal/logger/logger.go:136]: logger is not making a call to os.exit(1)

**Review by vadlakondaswetha — APPROVED** [review]

File: internal/logger/logger.go (hunk 4 above)
Comment (vadlakondaswetha) [reply internal/logger/logger.go:136]: also will it create any problem in doing os.exit(1) before the crash is happening?

**Pull request merged**